x-pack/plugins/index_lifecycle_management/public/application/services/policies/cold_phase.ts

```diff
@@ -53,9 +53,9 @@ export const coldPhaseFromES = (phaseSerialized?: SerializedColdPhase): ColdPhas
         Object.entries(allocate.require).forEach((entry) => {
           phase.selectedNodeAttrs = entry.join(':');
         });
-        if (allocate.number_of_replicas) {
-          phase.selectedReplicaCount = allocate.number_of_replicas.toString();
-        }
+      }
+      if (allocate.number_of_replicas !== undefined && allocate.number_of_replicas !== null) {
```

Comment:
It looks like this is the check we want to re-introduce: https://github.com/elastic/kibana/blob/fa51a8a9c3d1df6459ea7e8145044028fbbf375d/x-pack/plugins/index_lifecycle_management/public/application/store/selectors/policies.js#L174

I'd say we should simplify the check back to `!= null` as that covers both `null` and `undefined`

number_of_replicas can be set in allocate regardless of require, yes 👍🏻

Reply:
Same for warm of course 😁

Reply:
exactly, we need to check for both `null` and `undefined`. I'd prefer the strict comparison here as it's more verbose and then there's no need for a comment :)